packages/client/hmi-client/src/components/widgets/tera-chart-settings-panel.vue

```diff
@@ -114,7 +130,27 @@ const props = defineProps<{
 	generateAnnotation?: (setting: ChartSetting, query: string) => Promise<ChartAnnotation | null>;
 }>();
 
-const emit = defineEmits(['close', 'update-settings', 'delete-annotation', 'create-annotation']);
+type Items = { name: string; id: string; primaryColor: string };
+
+const variables = computed(() => {
+	const items: Items[] = [];
+	if (!props.activeSettings) return items;
+	const activeSettings = cloneDeep(props.activeSettings as ChartSettingComparison | null);
+	if (activeSettings?.type !== ChartSettingType.VARIABLE_COMPARISON) return [];
+	activeSettings.selectedVariables.forEach((value) => {
+		const item = { name: value, id: activeSettings.id, primaryColor: activeSettings.variableColors?.[value] ?? '' };
```

Comment:
Instead of modifying setting (`setting.variableColors = {};`), line 472 in `useChart.ts`, what if we just fall back to init `varaibleColors` with default CATEGORICAL_SCHEME values here if `variableColors` is undefined yet?